src/main/java/com/disraely/ioc/Ioc.java

```diff
@@ -0,0 +1,28 @@
+package com.disraely.ioc;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Ioc {
+    public static Ioc instance;
+    public static Map<String, Object> instances = new HashMap<>();
+
+    public Ioc() {
+    }
+
+    public static Ioc getInstance() {
+        if (instance == null) {
+            instance = new Ioc();
+        }
+        return instance;
+    }
+
+    public void add(String key, Object value) {
+        instances.put(key, value);
+    }
+
+    public <T> T get(String key) {
+        return (T) instances.get(key);
```

Comment:
what happens if the key does not exists? I think if you are looking for something here is because it should exist.

Reply:
Resolved!

Reply:
If key is null `KeyNotFoundException` does not seem to be pretty clear

Reply:
```
var validatedKey = Optional.ofNullable(key).filter(instances::containsKey).orElseThrow(...); 
return ... instances.get(validatedKey);
```